src/script.js

```diff
@@ -0,0 +1,231 @@
+const addTaskButton = document.querySelector(ADD_TASK_BTN_SELECTOR);
+const taskInput = document.querySelector(TASK_INPUT_SELECTOR);
+const allTasksContainer = document.querySelector(ALL_TASKS_CONTAINER_SELECTOR);
+const pendingTasksCounter = document.querySelector(TASKS_COUNTER_SELECTOR);
+const clearButton = document.querySelector(CLEAR_BTN_SELECTOR);
+const sortDownButton = document.querySelector(SORT_DOWN_BTN_SELECTOR);
+const sortUpButton = document.querySelector(SORT_UP_BTN_SELECTOR);
+const sortButtonsContainer = document.querySelector(SORT_BTNS_SELECTOR);
+const pokemonImagesContainer = document.querySelector(POKEMON_IMAGES_SELECTOR);
+const pokemonCatchText = document.querySelector(POKEMON_TEXT_SELECTOR);
+
+let pendingTasks = 0;
+
+const tasks = new ItemManager();
+const pokemonClient = new PokemonClient();
+
+addTaskButton.onclick = () => {
+   renderAndAnimate();
+};
+
+taskInput.onkeypress = (e) => {
+   if (e.key === "Enter") {
+      renderAndAnimate();
+   }
+};
+
+async function renderAndAnimate() {
+   if (await addTask()) {
+      const newTask = renderTasks();
+      createTaskAnimation(newTask);
+   }
+}
+
+clearButton.onclick = () => {
+   tasks.clear();
+   renderTasks();
+   pendingTasksUpdate(0);
+};
+
+sortDownButton.onclick = () => {
+   tasks.sortDown();
+   renderTasks();
+};
+
+sortUpButton.onclick = () => {
+   tasks.sortUp();
+   renderTasks();
+};
+
+function renderTasks() {
+   let lastTaskAdded;
+   while (allTasksContainer.firstChild) {
+      allTasksContainer.removeChild(allTasksContainer.lastChild);
+   }
+   tasks.items.forEach((item) => {
+      const taskContainer = createTaskContainer();
+      const newTask = createNewTask(taskContainer, item);
+      const deleteTask = createDeleteTaskButton(taskContainer);
+      addHoverReveal(taskContainer, deleteTask, newTask);
+      lastTaskAdded = newTask;
+   });
+
+   return lastTaskAdded;
+}
+
+async function addTask() {
+   const taskUserInput = taskInput.value;
+   taskInput.value = "";
+   const allPokemonNames = await pokemonClient.getAllPokemonNames();
+   if (taskUserInput.trim().length === 0) {
+      alert("Please fill in a task");
+
+      return false;
```

Comment:
this method is too long, try to extract to mini methods.
for example you can extract lines 70-73 to a method that performs validations

Reply:
What do you mean by extract to methods?  to functions? or to methods inside the item manager class?